test/lightweight_test_test.cpp

```diff
@@ -65,10 +65,15 @@ int main()
     BOOST_TEST_EQ( ++x, ++y );
     BOOST_TEST_EQ( x++, y++ );
 
+    const char* s1 = "abc"; // make sure addresses are different
+    const char* s2 = "abc"; // make sure addresses are different
```

Comment:
I see what you mean. I am open for suggestions on how to make it better/more portable :)

I tested this on gcc.godbolt.org and wandbox. It works ok with -O0.